### PR TITLE
Dimshuffle Layer

### PR DESCRIPTION
**Pull request by isaak-l**
Introduction of a Dimshuffle layer similar to the existing Reshape layer.

### CODE REVIEW

File: DeepFried2/layers/Dimshuffle.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import DeepFried2 as df
+
+
+class Dimshuffle(df.Module):
+    def __init__(self, *new_shape):
+        df.Module.__init__(self)
+        self.new_shape = new_shape
+
+    def symb_forward(self, symb_input):
+        return symb_input.dimshuffle(self.new_shape)
```
Comment (lucasb-eyer) [inline DeepFried2/layers/Dimshuffle.py]: Aren't you missing a `*` here?

`self.new_shape` is a tuple, but according to documentation, `dimshuffle` takes "splatted" arguments, not a tuple.

**Comment by lucasb-eyer** [comment]
I was also thinking about the name. Torch calls it [`permute`](https://github.com/torch/torch7/blob/master/doc/tensor.md#tensor-permutedim1-dim2--dimn) and Julia [`permutedims`](http://julia.readthedocs.org/en/latest/stdlib/arrays/#Base.permutedims).

Do you have any strong opinions?

**Comment by isaak-l** [comment]
We could call it something like permute. 
If the goal is to almost completely hide theano from the user than this might be a good idea.

However, if the user is somewhat familiar with theano dimshuffle should be clearer.

**Comment by lucasb-eyer** [comment]
Goal is the former, almost completely hide it, stress being more on "almost" than "completely". Let's go with `permute` then if you don't mind.

**Comment by lucasb-eyer** [comment]
lgtm now, squash & force-push and I'll merge.

**Comment by lucasb-eyer** [comment]
Thanks!

**Pull request merged**